### PR TITLE
Bugfix/product

### PR DESCRIPTION
**Pull request by ShahryarJahanshahloo**

### CODE REVIEW

File: src/Controller/Brand/BrandController.php [hunk 1]
```diff
@@ -2,15 +2,13 @@
 
 namespace App\Controller\Brand;
 
-use App\Entity\Brand\Brand;
 use App\Service\Brand\BrandManager;
```
Comment (Amirack2003) [inline src/Controller/Brand/BrandController.php:5]: Use Interface

File: src/Controller/Brand/BrandController.php [hunk 2]
```diff
@@ -2,95 +2,98 @@
...
+    public function getAll(): Response
     {
         try {
-            $body = $this->brandManager->getRequestBody($req);
-            $brand = $doctrine->getRepository(Brand::class)->findOneById($body['id']);
-            if (!$brand) return $this->json(['message' => "brand not found"], 400);
-            $message = $this->brandManager->removeUnused($brand);
-            if (array_key_exists('error', $message)) return $this->json(['message' => $message['error']], 400);
-            return $this->json(['message' => $message['message']]);
+            $brands = $this->brandManager->findALl();
```
Comment (do0ry) [inline src/Controller/Brand/BrandController.php:41]: edit ALl

File: src/Controller/Product/ProductController.php [hunk 3]
```diff
@@ -2,149 +2,138 @@
...
-    #[Route('/toggle-activity', name: 'toggle_activity', methods: ['PATCH'])]
-    public function toggleActivity(Request $req): Response
+    #[Route('/{id}/activity', name: 'toggle_activity', methods: ['PATCH'])]
+    public function toggleActivity(Request $req, int $id): Response
     {
         try {
             $body = $this->productManager->getRequestBody($req);
-            $message = $this->productManager->toggleActivity($body['id'], $body['active']);
-            if (array_key_exists('error', $message)) return $this->json(['message' => $message['error']], 400);
+            $message = $this->productManager->toggleActivity($id, $body['active']);
```
Comment (do0ry) [inline src/Controller/Product/ProductController.php:96]: notice --> variant /active/deactive

**Review by do0ry — APPROVED** [review]

File: src/Repository/Product/Filters.php [hunk 4]
```diff
@@ -11,27 +11,28 @@ class Filters
...
-            ->innerJoin(Variant::class, 'v', Join::WITH, 'p.id = v.product_id')
-            ->groupBy('p.id')
-            ->andWhere('p.active = 1')//or true
-            ->setFirstResult($options['offset'])
-            ->setMaxResults($options['limit']);
-        if ($options['availableOnly']) $qb->andWhere('v.quantity > 0');
-        if (array_key_exists('minPrice', $options)) $qb->andWhere('v.price > :minPrice')->setParameter('minPrice', $options['minPrice']);
-        if (array_key_exists('maxPrice', $options)) $qb->andWhere('v.price < :maxPrice')->setParameter('maxPrice', $options['maxPrice']);
-        $qb = self::addSortFilter($qb, $options['sortedBy']);
+        $qb->select('product.name, MIN(variant.price) AS price, product.viewCount, SUM(variant.soldNumber) as sold')
```
Comment (do0ry) [inline src/Repository/Product/Filters.php:14]: NICE :+1:

**Pull request merged**